Sources/CvUnit.cpp

```diff
@@ -7026,6 +7026,30 @@ bool CvUnit::canScrap() const
 }
 
 
+int CvUnit::calculateScrapValue() const
+{
+	float fCost = float(getUnitInfo().getProductionCost() * GC.getGameSpeedInfo(GC.getGame().getGameSpeedType()).getTrainPercent() / 100);
+	fCost += fCost * float(getUpgradeDiscount() / 100);
+
+	if (GC.getGame().isOption(GAMEOPTION_SIZE_MATTERS))
+	{
+		int iGroupDiff = groupRank() - m_pUnitInfo->getBaseGroupRank();
+		// Workaround because pow() seems to return 0 when having a negative exponent, for some reason??
+		// Otherwise just "fCost *= float(std::pow(3, iGroupDiff));" would work by itself
+		if (iGroupDiff >= 0)
+		{
+			fCost *= float(std::pow(3, iGroupDiff));
+		}
+		else
+		{
+			fCost *= float(1 / std::pow(3, -iGroupDiff));
+		}
```

Comment:
pow(int, int) returns a int doesn't it?
try changing it to:
 fCost *= 1 / std::pow(float(3), -iGroupDiff));